src/metrics.ts

```diff
@@ -182,33 +182,53 @@ export const MetricsProcessor = (
     };
   };
 
-  const _send = (): void => {
+  const _send = async () => {
     if (closed) {
       return;
     }
 
     const metrics: Metrics = _summarize();
     if (metrics) {
       log.debug('Start sending metrics data');
-      api
-        .postMetrics(environment, cluster, metrics)
-        .then((response) => {
-          log.debug('Metrics server returns: ', response.status);
-          infoMetricsSuccess(log);
-          if (response.status >= 400) {
-            log.error(
-              'Error while sending metrics data with status code: ',
-              response.status,
-            );
-          }
-        })
-        .catch((error: Error) => {
-          warnPostMetricsFailed(`${error}`, log);
-          log.debug('Metrics server returns error: ', error);
-        });
+      try {
+        const response = await postMetricsWithRetry(
+          environment,
+          cluster,
+          metrics,
+        );
+        log.debug('Metrics server returns:', response.status);
+        infoMetricsSuccess(log);
+      } catch (error) {
+        warnPostMetricsFailed(`${error}`, log);
```

Comment:
Does this need to be a template string, or is it just being used as a quick cast to string?